models/software_version.go

```diff
@@ -46,10 +60,37 @@ type SoftwareVersion struct {
 type SoftwareVersions []SoftwareVersion
 
 // Value implements database/sql/driver.Valuer interface. Should be defined on the value.
-func (sv SoftwareVersions) Value() (driver.Value, error) { return jsonValue(sv) }
+func (sv SoftwareVersions) Value() (driver.Value, error) {
+	for _, s := range sv {
+		if err := s.Name.Validate(); err != nil {
+			return nil, err
+		}
+
+		if s.Version == "" {
+			return nil, errors.Errorf("no version set for software %q", s.Name)
+		}
+	}
+	return jsonValue(sv)
+}
 
 // Scan implements database/sql.Scanner interface. Should be defined on the pointer.
-func (sv *SoftwareVersions) Scan(src interface{}) error { return jsonScan(sv, src) }
+func (sv *SoftwareVersions) Scan(src interface{}) error {
+	if err := jsonScan(sv, src); err != nil {
+		return err
+	}
+
+	for _, s := range *sv {
+		if err := s.Name.Validate(); err != nil {
```

Comment:
does it makes sense to validate in scan?

Reply:
no, removed